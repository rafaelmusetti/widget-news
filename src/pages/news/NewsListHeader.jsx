import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Select } from 'antd';

const { Option } = Select;

const NewsListHeaderContent = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 18px;
`;

const Title = styled.div`
  color: #009DFF;
  font-size: 32px;
  font-weight: 700;
  border-bottom: 4px solid #009DFF;
`;

/* istanbul ignore next */ const nook = () => {};
const NewsListHeader = ({
  onChangeFilter,
  sources,
}) => (
  <NewsListHeaderContent className="news-item">
    <Title>
      Not√≠cias
    </Title>
    <div>
      <Select
        allowClear
        onChange={onChangeFilter}
        placeholder="Filtrar por fonte"
        style={{ width: '170px' }}
      >
        {
          Object.keys(sources).map(source => (
            <Option value={source}>{source}</Option>
          ))
        }
      </Select>
    </div>
  </NewsListHeaderContent>
);

NewsListHeader.propTypes = {
  onChangeFilter: PropTypes.func,
  sources: PropTypes.arrayOf(PropTypes.shape({})),
};

NewsListHeader.defaultProps = {
  onChangeFilter: nook,
  sources: [],
};

export default NewsListHeader;
