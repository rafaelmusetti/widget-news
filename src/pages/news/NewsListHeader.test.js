import React from 'react';
import { shallow } from 'enzyme';
import NewsListHeader from './NewsListHeader';

describe('<NewsListHeader />', () => {
  describe('When rendering', () => {
    const tree = shallow(
      <NewsListHeader
        onChangeFilter={jest.fn()}
        sources={['test1', 'test2']}
      />,
    );

    it('should renders properly', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
