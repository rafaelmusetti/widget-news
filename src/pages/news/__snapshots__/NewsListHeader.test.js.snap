// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsListHeader /> When rendering should renders properly 1`] = `
<styled.div
  className="news-item"
>
  <styled.div>
    Not√≠cias
  </styled.div>
  <div>
    <Select
      allowClear={true}
      choiceTransitionName="zoom"
      onChange={[MockFunction]}
      placeholder="Filtrar por fonte"
      showSearch={false}
      style={
        Object {
          "width": "170px",
        }
      }
      transitionName="slide-up"
    >
      <Option
        value="0"
      >
        0
      </Option>
      <Option
        value="1"
      >
        1
      </Option>
    </Select>
  </div>
</styled.div>
`;
